frideos_library/lib/blocs/todos_bloc.dart

```diff
@@ -0,0 +1,167 @@
+import 'package:todos_repository_core/todos_repository_core.dart';
+
+import 'package:frideos/frideos_dart.dart';
+
+import 'package:frideos_library/models/models.dart';
+
+class TodosBloc {
+  final TodosRepository repository;
+
+  TodosBloc({this.repository}) {
+    print('-------TODOS BLOC INIT--------');
+    _init();
+  }
+
+  _init() async {
+    // Load the todos from the repository
+    await loadTodos();
+
+    // Listening for changes in the todos list, updated the streams and save
+    // to the repository. It is the same as:
+    //
+    // todosItems.outStream.listen((todos) => onTodosChange(...)
+    //
+    todosItems.onChange((todos) => onTodosChange(
+        allComplete, hasCompletedTodos, todos, onDataLoaded, todosSender));
+
+    // Listening for changes in the VisibilityFilter and filter the visible
+    // todos depending on the current filter.:
+    activeFilter
+        .onChange((filter) => onFilterChange(todosItems, visibleTodos, filter));
+  }
+
+  // When the data is loaded this function will save the todos, update the visible
+  // todos, and send to the statsBloc the todos list.
+  Function onDataLoaded;
```

Comment:
Why not just use a normal method here instead of defining this function in the constructor? E.g.

```
void onDataLoaded() {
    saveTodos();
    updateVisibleItems();
    todosSender.send(todosItems.value);
  }
```

Reply:
My bad, that was supposed to be cleaned before sending the PR, as the print on the blocs dispose method. The issue with the integration tests made me forget to clean these things up. Sorry.